src/Command/Environment/EnvironmentPushCommand.php

```diff
@@ -0,0 +1,111 @@
+<?php
+namespace Platformsh\Cli\Command\Environment;
+
+use Platformsh\Cli\Command\CommandBase;
+use Platformsh\Cli\Exception\RootNotFoundException;
+use Symfony\Component\Console\Input\InputArgument;
+use Symfony\Component\Console\Input\InputInterface;
+use Symfony\Component\Console\Input\InputOption;
+use Symfony\Component\Console\Output\OutputInterface;
+
+class EnvironmentPushCommand extends CommandBase
+{
+
+    protected function configure()
+    {
+        $this
+            ->setName('environment:push')
+            ->setAliases(['push'])
+            ->setDescription('Push code to an environment')
+            ->addArgument('src', InputArgument::OPTIONAL, 'The source ref', 'HEAD')
+            ->addOption('force', null, InputOption::VALUE_NONE, 'Force push')
+            ->addOption('dry-run', null, InputOption::VALUE_NONE, 'Dry run')
+            ->addOption('no-wait', null, InputOption::VALUE_NONE, 'After pushing, do not wait for build or deploy')
+            ->addOption('identity-file', null, InputOption::VALUE_REQUIRED, 'Specify an SSH identity file (public key) to use');
+        $this->addProjectOption()
+            ->addEnvironmentOption();
+    }
+
+    protected function execute(InputInterface $input, OutputInterface $output)
+    {
+        $this->validateInput($input);
+        $projectRoot = $this->getProjectRoot();
+        if (!$projectRoot) {
+            throw new RootNotFoundException();
+        }
+
+        $project = $this->getSelectedProject();
+        $environment = $this->getSelectedEnvironment();
+        $source = $input->getArgument('src');
+        $gitUrl = $project->getGitUrl();
+
+        /** @var \Platformsh\Cli\Helper\ShellHelper $shellHelper */
+        $shellHelper = $this->getHelper('shell');
+
+        if ($source && strpos($source, ':') !== false) {
+            $this->stdErr->writeln('Invalid ref: ' . $source);
+            return 1;
+        }
+
+        $command = sprintf(
+            'git push %s %s:%s',
+            escapeshellarg($gitUrl),
+            escapeshellarg($source),
+            escapeshellarg($environment->id)
+        );
+
+        foreach (['force', 'dry-run'] as $option) {
+            if ($input->getOption($option)) {
+                $command .= ' --' . $option;
+            }
+        }
+
+        $sshOptions = [];
+        if ($input->getOption('no-wait')) {
+            $sshOptions[] = "-o 'SendEnv PLATFORMSH_PUSH_*'";
+            $command = 'PLATFORMSH_PUSH_NO_WAIT=1 ' . $command;
+        }
+        if ($identityFile = $input->getOption('identity-file')) {
+            if (!file_exists($identityFile)) {
+                $this->stdErr->writeln('File not found: ' . $identityFile);
+                return 1;
+            }
+            $sshOptions[] = '-i ' . escapeshellarg($identityFile);
+        }
+
+        $tempFile = null;
+        if (!empty($sshOptions)) {
+            $sshCommand = 'ssh ' . implode(' ', $sshOptions) . ' $*';
```

Comment:
Small thing: I'd add a newline at the end... for esthetic purposes. Yes, it's a temporary file. I have a problem.